content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,351 @@
+model Post {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  title  String
+  userId String @db.ObjectId
+  user   User   @relation(fields: [userId], references: [id])
+}
+
+model User {
+  id    String @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  posts Post[]
+}
+```
+
+For more information on how to use relations in Prisma, see [our documentation](/concepts/components/prisma-schema/relations).
+
+### How to filter for <inlinecode>null</inlinecode> and undefined fields
+
+To understand how MongoDB distinguishes between `null` and undefined fields, consider the example of a `User` model with an optional `name` field:
+
+```ts
+model User {
+  id    String  @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  name  String?
+}
+```
+
+First, try creating a record with the `name` field explicitly set to `null`. Prisma will return `name: null` as expected:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+const createNull = await prisma.user.create({
+  data: {
+    email: 'user1@prisma.io',
+    name: null,
+  },
+})
+console.log(createNull)
+```
+
+</cmd>
+
+<cmdResult>
+
+```
+{
+    id: '6242c4ae032bc76da250b207',
+    email: 'user1@prisma.io',
+    name: null
+}
+```
+
+</cmdResult>
+
+</CodeWithResult>
+
+If you check your MongoDB database directly, you will also see a new record with `name` set to `null`:
+
+```terminal
+_id: 6242c4af032bc76da250b207
+email: "user1@prisma.io"
+name: null
+```
+
+Next, try creating a record without explicitly setting the `name` field:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+  const createUndefined = await prisma.user.create({
+    data: {
+       email: "user2@prisma.io",
+     }
+   })
+   console.log(createUndefined)
+}
+```
+
+</cmd>
+
+<cmdResult>
+
+```
+{
+    id: '6242c4ae032bc76da250b208',
+    email: 'user2@prisma.io',
+    name: null
+},
+
+```
+
+</cmdResult>
+
+</CodeWithResult>
+
+Prisma still returns `name: null`, but if you look in the database directly you will see that the record has no `name` field defined at all:
+
+```terminal
+_id: 6242c4af032bc76da250b208
+email: "user2@prisma.io"
+```
+
+Prisma returns the same result in both cases, because we currently don't have a way to specify this difference in MongoDB between fields that are `null` in the underlying database, and fields that are not defined at all — see [this Github issue](https://github.com/prisma/prisma/issues/12555) for more information.
+
+This means that you currently have to be careful when filtering for `null` and undefined fields. Filtering for records with `name: null` will only return the first record, with the `name` explicitly set to `null`:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+const findNulls = await prisma.user.findMany({
+  where: {
+    name: null,
+  },
+})
+console.log(findNulls)
+```
+
+</cmd>
+
+<cmdResult>
+
+```terminal
+[
+  {
+    id: '6242c4ae032bc76da250b207',
+    email: 'user1@prisma.io',
+    name: null
+  }
+]
```

Comment:
This is the first one where the terminal output is in an array and not just the object - should _all_ the ones above also have that actually?

Reply:
My memory is that the two `findMany`s should return an array, whereas the `create`s above shouldn't